models/DataObject/Objectbrick/Definition/Dao.php

```diff
@@ -173,4 +177,22 @@ public function classSaved(DataObject\ClassDefinition $classDefinition)
 
         $this->handleEncryption($classDefinition, [$tableQuery, $tableStore]);
     }
+
+    /**
+     * @param string $table
+     * @param array $columnsToRemove
+     * @param array $protectedColumns
+     */
+    protected function removeIndices($table, $columnsToRemove, $protectedColumns)
+    {
+        if (is_array($columnsToRemove) && count($columnsToRemove) > 0) {
+            $indexPrefix = str_starts_with($table, 'object_brick_query_') ? 'p_index_' : 'u_index_';
+            foreach ($columnsToRemove as $value) {
+                if (!in_array(strtolower($value), array_map('strtolower', $protectedColumns))) {
```

Comment:
I don't think you should call this `array_map('strtolower', $protectedColumns)` over and over again in the loop.

Reply:
Good point, I copied this part of the method. I think we should do it for both then.